File: backend/java/elo-calculator-backend/src/main/java/de/iits/elocalculatorbackend/player/model/dto/PlayerCreateOrUpdateRequestDto.java
package de.iits.elocalculatorbackend.player.model.dto;

import de.iits.elocalculatorbackend.player.model.entity.Player;

public record PlayerCreateOrUpdateRequestDto(
        String username,
        String displayName,
        String email
) {
    public Player toEntity() {
        Player player = new Player();
        player.setUsername(username);
        player.setDisplayName(displayName);
        player.setEmail(email);
        return player;
    }
}

File: backend/kotlin/src/main/kotlin/de/iits/elo/player/model/dto/PlayerCreateOrUpdateRequestDto.kt
package de.iits.elo.player.model.dto

import de.iits.elo.player.model.entity.Player

data class PlayerCreateOrUpdateRequestDto(

    val username: String,
    val displayName: String,
    val email: String?,
)

fun PlayerCreateOrUpdateRequestDto.toEntity() =
    Player(
        username = username,
        displayName = displayName,
        email = email,
    )
